examples/UART/echo.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * echo.cpp
+ *
+ *  Created on: Jan. 26, 2024
+ *      Author: manoj
+ */
+
+
+
+#include "phal_uart.h"
+
+#define MYSERIAL_INTR_PRIORITY_L  126 /*use only 2-126 and make sure not to overlap*/
+#define MYSERIAL_INTR_PRIORITY_H  125 /*use only 2-126 and make sure not to overlap*/
+
+uart_port_id mySerial = PHAL_UART_SCI_PORT;
+uart_intr_setup mySerialIntr = {.priority_high = MYSERIAL_INTR_PRIORITY_H,
+                                .priority_low = MYSERIAL_INTR_PRIORITY_L,
+                                .rx_level = PHAL_UART_INTR_HIGH,
+                                .tx_level = PHAL_UART_INTR_LOW};
+
+volatile uint32 tx_success = 0;
+volatile uint32 rx_success = 0;
+volatile uint8 message;
+
+char* message1 = "A for Effort!\n\r";
+
+void rxCallback(char data) {
+    if (rx_success)
```

Comment:
Likely related to my comment about having async rx/tx, but this check should probably be done in your driver.

Reply:
Removed the need for flags https://github.com/sfuphantom/hercules-phantom-lib/pull/13/commits/47fca4fdda2f1405f9866ef8b6836055d06a2ead